python/lsst/qa/explorer/functors.py

```diff
@@ -767,3 +793,169 @@ class Seeing(Functor):
+        super().__init__(**kwargs)
+
+    def computeReScalar(xx, yy, xy):
+        quad = afwGeom.Quadrupole(xx, yy, xy)
+        axes = afwGeom.ellipses.Axes(quad)
+        return np.sqrt(axes.getA() * axes.getB())
+
+    computeReVec = np.vectorize(computeReScalar)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        reArr = self.computeReVec(df[self.colXX], df[self.colYY], df[self.colXY])
+        return pd.Series(reArr, index=df.index)
+
+
+class ReferenceBand(Functor):
+    name = 'Reference Band'
+    shortname = 'refBand'
+
+    @property
+    def columns(self):
+        return ["merge_measurement_i",
+                "merge_measurement_r",
+                "merge_measurement_z",
+                "merge_measurement_y",
+                "merge_measurement_g"]
+
+    def _func(self, df):
+        def getFilterAliasName(row):
+            # get column name with the max value (True > False)
+            colName = row.idxmax()
+            return colName.replace('merge_measurement_', '')
+
+        return df[self.columns].apply(getFilterAliasName, axis=1)
+
+
+class Photometry(Functor):
+    AB_FLUX_SCALE = 3.630780547701013425e12  # dn to NanoJansky
+    LOG_AB_FLUX_SCALE = 12.56
+    FIVE_OVER_2LOG10 = 1.085736204758129569
+    COADD_ZP = 27
+
+    def __init__(self, colFlux, colFluxErr=None, calib=None, **kwargs):
+        self.vhypot = np.vectorize(self.hypot)
+        self.col = colFlux
+        self.colFluxErr = colFluxErr
+
+        self.calib = calib
+        if calib is not None:
+            self.fluxMag0, self.fluxMag0Err = calib.getFluxMag0()
+        else:
+            self.fluxMag0 = 1./np.power(10, -0.4*self.COADD_ZP)
+            self.fluxMag0Err = 0.
+
+        super().__init__(**kwargs)
+
+    @property
+    def columns(self):
+        return [self.col]
+
+    @property
+    def name(self):
+        return 'mag_{0}'.format(self.col)
+
+    @classmethod
+    def hypot(cls, a, b):
+        if np.abs(a) < np.abs(b):
+                a, b = b, a
+        if a == 0.:
+            return 0.
+        q = b/a
+        return np.abs(a) * np.sqrt(1. + q*q)
+
+    def dn2flux(self, dn, fluxMag0):
+        return self.AB_FLUX_SCALE * dn / fluxMag0
+
+    def dn2mag(self, dn, fluxMag0):
+        with np.warnings.catch_warnings():
+            np.warnings.filterwarnings('ignore', r'invalid value encountered')
+            np.warnings.filterwarnings('ignore', r'divide by zero')
+            return -2.5 * np.log10(dn/fluxMag0)
+
+    def dn2fluxErr(self, dn, dnErr, fluxMag0, fluxMag0Err):
+        retVal = self.vhypot(dn * fluxMag0Err, dnErr * fluxMag0)
+        retVal *= self.AB_FLUX_SCALE / self.fluxMag0 / self.fluxMag0
+        return retVal
+
+    def dn2MagErr(self, dn, dnErr, fluxMag0, fluxMag0Err):
+        retVal = self.dn2fluxErr(dn, dnErr, fluxMag0, fluxMag0Err) / self.dn2flux(dn, self.fluxMag0)
+        return self.FIVE_OVER_2LOG10 * retVal
```

Comment:
I assume you're not using `Calib` here so you can make use of `meas_mosaic`/`jointcal` outputs before `PhotoCalib` fully arrives?

Reply:
Just some sloppiness, but yes assuming we don't have access to the Calibs at this stage, and we'll load in a spatially varying ZP per object at the writeObjectTable stage.